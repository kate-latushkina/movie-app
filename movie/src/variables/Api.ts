// import { Request } from "./Request";

const API_KEY = "f37ea435a74a07c3e824148f66ca3fc7";
const API_URL = `https://api.themoviedb.org/3/discover/movie?sort_by=popularity.desc&api_key=${API_KEY}&page=1`;
const API_IMG = "https://image.tmdb.org/t/p/original";

export { API_KEY, API_URL, API_IMG };

// export const { addMovies, sentMovie, putMovie, deleteMovie } = {
//   addMovies: Request.getRequest(API_URL),
//   sentMovie: Request.postRequest(API_URL, ),
//   deleteMovie: Request.deleteRequest(API_URL),
//   putMovie: Request.putRequest(API_URL),
// };
