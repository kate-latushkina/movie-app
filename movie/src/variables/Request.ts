class Requests {
  handleResponse = (res: Response) =>
    !res.ok ? Promise.reject(res) : res.json();

  handleFetch = async (url: string, params: { [method: string]: string }) => {
    const data = await fetch(url, params)
      .then(res => {
        return this.handleResponse(res);
      })
      .catch(err => console.warn(`Error fetching data from ${url}, ${err}`));

    if (params.method === "GET") {
      return data.results;
    }
    return data;
  };

  getRequest = async (url: string) => {
    const params = {
      method: "GET",
    };

    return this.handleFetch(url, params);
  };

  postRequest = async (url: string, body: { [body: string]: string }) => {
    const params = {
      method: "POST",
      body: JSON.stringify({
        ...body,
      }),
    };

    return this.handleFetch(url, params);
  };

  putRequest = async (url: string, body: { [body: string]: string }) => {
    const params = {
      method: "PUT",
      body: JSON.stringify({
        ...body,
      }),
    };

    return this.handleFetch(url, params);
  };

  deleteRequest = async (url: string, body: { [body: string]: string }) => {
    const params = {
      method: "DELETE",
      body: JSON.stringify({
        ...body,
      }),
    };

    return this.handleFetch(url, params);
  };
}

const Request = new Requests();

export { Request };
