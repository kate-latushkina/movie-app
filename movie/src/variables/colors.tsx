const colors = {
  darkViolet: "#593158",
  darkGrey: "#424242",
  lightGrey: "#555555",
  pink: "#f65261",
  white: "#fff",
};

export default colors;
