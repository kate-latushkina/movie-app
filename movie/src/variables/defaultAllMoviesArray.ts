const defaultAllMoviesArray = [
  {
    id: 0,
    title: "",
    overview: "",
    poster_path: "",
    release_date: "2021-01-01",
    genre_ids: [1],
    vote_average: 5,
  },
];

export default defaultAllMoviesArray;
