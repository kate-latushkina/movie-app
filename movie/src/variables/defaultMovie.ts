const defaultValueForm = {
  id: "0",
  title: "",
  release_date: "2021-01-01",
  movieUrl: "",
  genre: "",
  overview: "",
  vote_average: 0,
};

export default defaultValueForm;
