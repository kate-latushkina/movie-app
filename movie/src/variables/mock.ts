export const CARD_PROP_MOC = {
  id: 1,
  overview: "This is description",
  poster_path: "/9kg73Mg8WJKlB9Y2SAJzeDKAnuB.jpg",
  title: "Poster",
  vote_average: 4,
  release_date: "2021",
};
