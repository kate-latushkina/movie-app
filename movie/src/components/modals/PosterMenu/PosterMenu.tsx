import { createStyles, makeStyles } from "@material-ui/core";
import React, { useCallback, useState } from "react";
import dotsImage from "../../../img/ellipsis-v-solid.svg";
import colors from "../../../variables/colors";

const useStyles = makeStyles(() =>
  createStyles({
    dotsMenu: {
      position: "absolute",
      top: "10px",
      right: "10px",
      backgroundColor: "initial",
      border: "none",
    },
    dotsImageBlock: {
      height: "25px",
      width: "25px",
      padding: "5px",
      border: `1px solid ${colors.darkGrey}`,
      borderRadius: "50%",
      backgroundColor: colors.darkGrey,
      cursor: "pointer",
    },
    dotsListBlock: {
      width: "100px",
      padding: "0 0 10px",
      margin: 0,
      position: "absolute",
      right: 0,
      top: "10px",
      listStyle: "none",
      backgroundColor: colors.darkGrey,
      color: colors.white,
    },
    closeButton: {
      backgroundColor: "initial",
      marginLeft: "75%",
      border: "none",
      cursor: "pointer",
      color: colors.white,
    },
    dotsList: {
      cursor: "pointer",
      padding: "5px",
    },
    dotListDelete: {
      "&:hover": {
        backgroundColor: "red",
      },
    },
  }),
);

type IPosterMenuProps = {
  id: number;
};

const PosterMenu: React.FC<IPosterMenuProps> = ({ id }: IPosterMenuProps) => {
  const classes = useStyles();
  const [isPosterMenuListShow, setPosterMenuListShow] = useState(false);

  const PosterMenuList: React.FC = () => {
    const onClose = useCallback(() => {
      setPosterMenuListShow(false);
    }, [isPosterMenuListShow]);
    console.log("PosterMenu", id);

    return (
      <ul className={classes.dotsListBlock}>
        <button type="button" className={classes.closeButton} onClick={onClose}>
          X
        </button>
        <li className={classes.dotsList}>Edit</li>
        <li className={`${classes.dotsList} ${classes.dotListDelete}`}>
          Delete
        </li>
      </ul>
    );
  };

  const handleClick = useCallback(() => {
    setPosterMenuListShow(true);
  }, [isPosterMenuListShow]);

  return (
    <>
      <button type="button" onClick={handleClick} className={classes.dotsMenu}>
        <img src={dotsImage} alt="dots" className={classes.dotsImageBlock} />
      </button>
      {isPosterMenuListShow ? <PosterMenuList /> : null}
    </>
  );
};

export default PosterMenu;
