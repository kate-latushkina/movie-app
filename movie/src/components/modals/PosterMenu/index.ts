import PosterMenu from "./PosterMenu";

export default PosterMenu;
