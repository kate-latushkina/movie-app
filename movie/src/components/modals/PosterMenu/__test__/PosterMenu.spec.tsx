import React from "react";
import toJSON from "enzyme-to-json";
import { mount, ReactWrapper } from "enzyme";
import PosterMenu from "../PosterMenu";
import { CARD_PROP_MOC } from "../../../../variables/mock";

describe("<PosterMenu />", () => {
  let wrapper: ReactWrapper;
  beforeEach(() => {
    wrapper = mount(<PosterMenu id={CARD_PROP_MOC.id} />);
  });
  it("should mount", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
