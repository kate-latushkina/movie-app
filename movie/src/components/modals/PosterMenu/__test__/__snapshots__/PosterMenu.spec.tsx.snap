// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PosterMenu /> should mount 1`] = `
<PosterMenu
  id={1}
>
  <button
    className="makeStyles-dotsMenu-1"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="dots"
      className="makeStyles-dotsImageBlock-2"
      src="ellipsis-v-solid.svg"
    />
  </button>
</PosterMenu>
`;
